pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -210,15 +225,48 @@ type ClusterWorkspaceTypeSpec struct {
 	AllowedParentWorkspaceTypes []ClusterWorkspaceTypeName `json:"allowedParentWorkspaceTypes,omitempty"`
 }
 
+// ClusterWorkspaceTypeExtension defines how other ClusterWorkspaceTypes are
+// composed together to add functionality to the owning ClusterWorkspaceType.
+type ClusterWorkspaceTypeExtension struct {
+	// with are ClusterWorkspaceTypes whose initializers are added to the list
+	// for the owning type, and for whom the owning type becomes an alias.
+	//
+	// +optional
+	With []ClusterWorkspaceTypeReference `json:"with,omitempty"`
+
+	// without are ClusterWorkspaceTypes whose initializers are removed from the list
+	// computed for "with", and for whom the owning type does not become an alias.
+	//
+	// +optional
+	Without []ClusterWorkspaceTypeReference `json:"without,omitempty"`
+}
+
 // These are valid conditions of ClusterWorkspaceType.
 const (
 	ClusterWorkspaceTypeVirtualWorkspaceURLsReady conditionsv1alpha1.ConditionType = "VirtualWorkspaceURLsReady"
+	ClusterWorkspaceTypeExtensionsResolved        conditionsv1alpha1.ConditionType = "ExtensionsResolved"
 
-	ErrorGeneratingURLsReason = "ErrorGeneratingURLs"
+	ErrorGeneratingURLsReason      = "ErrorGeneratingURLs"
+	ErrorResolvingExtensionsReason = "ErrorResolvingExtensionsReason"
 )
 
 // ClusterWorkspaceTypeStatus defines the observed state of ClusterWorkspaceType.
 type ClusterWorkspaceTypeStatus struct {
+	// Initializers are the list of initializing controller references which need
```

Comment:
Sorry, missed this before. All field-level godoc in k8s API types must begin with a lowercase letter (contrary to standard godoc practices). Global comment